server/handler/uast_integration_test.go

```diff
@@ -2,19 +2,156 @@ package handler_test
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 	"testing"
 
+	"github.com/go-chi/chi"
 	"github.com/pressly/lg"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/suite"
+	"gopkg.in/src-d/enry.v1"
 
 	"github.com/src-d/gitbase-web/server/handler"
 	"github.com/src-d/gitbase-web/server/serializer"
+	"github.com/src-d/gitbase-web/server/service"
 )
 
+type UASTGetLanguagesSuite struct {
+	suite.Suite
+	handler http.Handler
+}
+
+func TestUASTGetLanguagesSuite(t *testing.T) {
+	if !isIntegration() {
+		t.Skip("use the env var GITBASEPG_INTEGRATION_TESTS=true to run this test")
+	}
+
+	q := new(UASTGetLanguagesSuite)
+	r := chi.NewRouter()
+	r.Use(lg.RequestLogger(logrus.New()))
+	r.Post("/detect-lang", handler.APIHandlerFunc(handler.DetectLanguage()))
+	r.Get("/get-languages", handler.APIHandlerFunc(handler.GetLanguages(bblfshServerURL())))
+
+	q.handler = r
+
+	suite.Run(t, q)
+}
+
+func UnmarshalGetLanguagesResponse(b []byte) []service.Language {
+	var resBody struct {
+		Data []service.Language `json:"data"`
+	}
+	json.Unmarshal(b, &resBody)
+	return resBody.Data
+}
+
+func (suite *UASTGetLanguagesSuite) TestSameEnryLanguage() {
+	require := suite.Require()
+
+	req, _ := http.NewRequest("GET", "/get-languages", strings.NewReader(""))
+
+	res := httptest.NewRecorder()
+	suite.handler.ServeHTTP(res, req)
+
+	require.Equal(http.StatusOK, res.Code, res.Body.String())
+
+	escapeForJSON := func(s string) string {
+		return strings.Replace(strings.Replace(s, "\"", "\\\"", -1),
+			"\n", "\\n", -1)
+	}
+
+	for _, lang := range UnmarshalGetLanguagesResponse(res.Body.Bytes()) {
+		langName := lang.Name
+		suite.T().Run(langName, func(t *testing.T) {
+			content, filename := suite.getContentAndFilename(langName)
+			jsonRequest := fmt.Sprintf(`{ "content": "%s", "filename": "%s" }`,
+				escapeForJSON(content), filename)
+			req, _ := http.NewRequest("POST", "/detect-lang", strings.NewReader(jsonRequest))
+
+			res = httptest.NewRecorder()
+			suite.handler.ServeHTTP(res, req)
+
+			require.Equal(http.StatusOK, res.Code, res.Body.String())
+
+			detectedLang, detectedLangType := handler.UnmarshalDetectLangResponse(res.Body.Bytes())
+
+			if langName == "Bash" {
+				require.NotEqual(langName, detectedLang)
+				t.Skip("TEST FAILURE IS A KNOWN ISSUE")
```

Comment:
Is this correct? Isn't this due to the aliasing problem that we also have on engine? But shouldn't it be true also for `C#`?